src/software/ai/passing/pass_evaluation_test.cpp

```diff
@@ -0,0 +1,102 @@
+#include "software/ai/passing/pass_evaluation.h"
+
+#include <gtest/gtest.h>
+
+#include <unordered_map>
+
+#include "software/ai/passing/eighteen_zone_pitch_division.h"
+#include "software/ai/passing/pass.h"
+#include "software/geom/algorithms/contains.h"
+#include "software/world/field.h"
+
+TEST(PassEvaluation, best_pass_over_entire_field)
+{
+    auto field          = Field::createSSLDivisionBField();
+    auto pitch_division = std::make_shared<const EighteenZonePitchDivision>(field);
+
+    std::unordered_map<EighteenZoneId, PassWithRating> passes_with_rating;
+
+    for (EighteenZoneId zone_id : allValuesEighteenZoneId())
+    {
+        passes_with_rating.emplace(zone_id,
+                                   PassWithRating{std::move(Pass(Point(0, 0), 0)), 0});
+    }
+
+    // change zone 8 to have a score of 0.8, which should be the highest
+    passes_with_rating.emplace(EighteenZoneId::ZONE_8,
+                               PassWithRating{std::move(Pass(Point(0, 0), 0)), 0.8});
+
+    auto pass_eval = PassEvaluation<EighteenZoneId>(
+        pitch_division, passes_with_rating, std::make_shared<const PassingConfig>(),
+        Timestamp::fromSeconds(10));
+
+    ASSERT_EQ(pass_eval.getBestPassOnField(),
+              passes_with_rating.at(EighteenZoneId::ZONE_8));
+}
+
+
+TEST(PassEvaluation, best_pass_in_zones)
+{
+    auto field          = Field::createSSLDivisionBField();
+    auto pitch_division = std::make_shared<const EighteenZonePitchDivision>(field);
+
+    std::unordered_map<EighteenZoneId, PassWithRating> passes_with_rating;
+
+    for (EighteenZoneId zone_id : allValuesEighteenZoneId())
+    {
+        passes_with_rating.emplace(zone_id,
+                                   PassWithRating{std::move(Pass(Point(0, 0), 0)),
+                                                  0.05 * static_cast<unsigned>(zone_id)});
```

Comment:
shouldn't the passes be _in_ each of the zones?

Reply:
no, they can leave the zones

the pass generator makes an effort to the keep the passes in the zone so we don't clump our robots up together, but it doesn't explicitly prevent them from leaving

on an actual world, your passes will be generally in the zone or near the zone boundary, but its not enforced by the pass evaluation

if we wanted to restrict the pass to the zone, we would do this in the pass generator

Reply:
can we document this somehow?

Reply:
Its already documented in `pass_generator.h` 

````
     * Creates a new PassGenerator with the given pitch_division.                                                                                                                                                                                                                 
     *                                                                                                                                                                                                                                                                            
     * The PassGenerator will use this pitch division to guide initial random samples                                                                                                                                                                                             
     * in each zone after the pitch has been divided.                                                                                                                                                                                                                             
```